pkg/rehearse/jobs_test.go

```diff
@@ -144,33 +144,43 @@ func makeTestingPresubmit(name, context string, ciopArgs []string) *prowconfig.P
 				}},
 			},
 		},
-		Context:  context,
-		Brancher: prowconfig.Brancher{Branches: []string{"^master$"}},
+		Context: context,
+		Brancher: prowconfig.Brancher{Branches: []string{
+			fmt.Sprintf("^%s$", branch),
+		}},
 	}
 }
 
 func TestMakeRehearsalPresubmit(t *testing.T) {
+	testPrNumber := 123
 	testCases := []struct {
-		source   *prowconfig.Presubmit
-		pr       int
-		expected *prowconfig.Presubmit
+		description string
+		source      *prowconfig.Presubmit
+		expected    *prowconfig.Presubmit
 	}{{
-		source: makeTestingPresubmit("pull-ci-openshift-ci-operator-master-build", "ci/prow/build", []string{"arg", "arg"}),
-		pr:     123,
+		description: "sanity test",
+		source: makeTestingPresubmit(
+			"pull-ci-openshift-ci-operator-master-build",
+			"ci/prow/build",
+			[]string{"arg", "arg"},
+			"master"),
 		expected: makeTestingPresubmit(
```

Comment:
Using helpers like `makeTestingPresubmit` make the tests more dependent on each other and fragile. I think unless we have a good reason we should use literals where possible. This is not something you need to tackle in this PR

Reply:
I already started rewriting that once but did not finish because I got bogged down in changes. 

I'd like to avoid crafting `Presubmit` literals everywhere - they are quite deep structures, it's easy to make mistakes because stuff needs to match in them, and most tests really do not care about most fields in them, they just need "sane" structs.

At the same time, what you said is valid and the current state is certainly not ideal.